### PR TITLE
istioctl: move commands to istioctl/pkg, and fix dependency issues

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by hanxiaop** [comment]
Part of https://github.com/istio/istio/issues/38503

File: istioctl/cmd/istioctl_test.go [hunk 1]
```diff
@@ -15,34 +15,18 @@
...
 import (
-	"bytes"
-	"regexp"
-	"strings"
 	"testing"
 
-	"github.com/spf13/cobra"
-
-	"istio.io/istio/pilot/test/util"
+	util2 "istio.io/istio/istioctl/pkg/util"
```
Comment (howardjohn) [inline istioctl/cmd/istioctl_test.go:20]: can we cleanup the util2

File: istioctl/pkg/config/config.go [hunk 2]
```diff
@@ -74,11 +75,15 @@ func runList(writer io.Writer) error {
 		i++
 	}
 	sort.Strings(keys)
-
+	ccc := settableFlags
+	fmt.Println(ccc)
```
Comment (howardjohn) [inline istioctl/pkg/config/config.go:79]: drop this and L85

File: istioctl/cmd/testdata/vmconfig/.gitignore
Comment (howardjohn) [inline istioctl/cmd/testdata/vmconfig/.gitignore:1]: can we add this back? I see these files after running tests:
```
        istioctl/cmd/testdata/vmconfig-nil-proxy-metadata/
        istioctl/cmd/testdata/vmconfig/
```

Comment (howardjohn) [reply istioctl/cmd/testdata/vmconfig/.gitignore:1]: or equivilent

File: istioctl/pkg/proxyconfig/proxyconfig.go [hunk 3]
```diff
@@ -29,6 +29,9 @@ import (
 	"sigs.k8s.io/yaml"
 
 	"istio.io/istio/istioctl/pkg/cli"
+	"istio.io/istio/istioctl/pkg/completion"
+	"istio.io/istio/istioctl/pkg/kubeinject"
+	util2 "istio.io/istio/istioctl/pkg/util"
```
Comment (howardjohn) [inline istioctl/pkg/proxyconfig/proxyconfig.go:34]: nit: avoid util2, either `istioctlutil` or something?

File: istioctl/pkg/proxyconfig/proxyconfig_test.go [hunk 4]
```diff
@@ -12,18 +12,24 @@
...
 	"fmt"
+	"net/http"
 	"strings"
 	"testing"
 
 	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest/fake"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	util2 "k8s.io/kubectl/pkg/cmd/util"
```
Comment (howardjohn) [inline istioctl/pkg/proxyconfig/proxyconfig_test.go:28]: avoid util2

File: istioctl/pkg/revision/revision.go [hunk 5]
```diff
@@ -35,7 +34,9 @@ import (
 
 	"istio.io/api/label"
 	"istio.io/istio/istioctl/pkg/cli"
+	"istio.io/istio/istioctl/pkg/root"
 	"istio.io/istio/istioctl/pkg/tag"
+	util2 "istio.io/istio/istioctl/pkg/util"
```
Comment (howardjohn) [inline istioctl/pkg/revision/revision.go:39]: nit: rename util2

**Review by howardjohn — APPROVED** [review]
wow awesome! a few nits, only blocking comment is the printlns that slipped in. Ill approve now though since UX needs to approve as well.

File: istioctl/pkg/config/config.go (hunk 2 above)
Comment (hanxiaop) [reply istioctl/pkg/config/config.go]: Thanks, fixed

File: istioctl/cmd/istioctl_test.go (hunk 1 above)
Comment (hanxiaop) [reply istioctl/cmd/istioctl_test.go]: Thanks, all the `util2` usages have been refactored to `istioctlutil`

File: istioctl/cmd/testdata/vmconfig/.gitignore
Comment (hanxiaop) [reply istioctl/cmd/testdata/vmconfig/.gitignore:1]: I think I forgot moving this to the right directory which should be `istioctl/pkg/workload/testdata`, thanks

Comment (hanxiaop) [reply istioctl/cmd/testdata/vmconfig/.gitignore:1]: This is related to the test cleanup logic. Every time I switch to a different branch and run tests, this occurs. I just submitted a PR for this at https://github.com/istio/istio/pull/45478.

**Review by GregHanson — APPROVED** [review]

**Pull request merged**